website/blog/2021-05-17-codepush.md

```diff
@@ -0,0 +1,409 @@
+
+```javascript
+import codePush from "react-native-code-push";
+
+let App: () => React$Node = () => {
+    //your app component
+}
+
+App = codePush({ checkFrequency: codePush.CheckFrequency.MANUAL })(App);
+
+export default App;
+```
+
+Then you can use use the `sync()` function provided by the CodePush APIs to manually trigger the update check. You can call this function, for example, when a button is pressed:
+
+```javascript
+const onCheckUpdates = async () => {
+  var result = await codePush.sync({
+      updateDialog: true,
+      installMode: codePush.InstallMode.IMMEDIATE
+  });
+}
+```
+
+### Track the download progress
+
+CodePush provides a series of events to track the status of the installation. The only caveat is that, at the moment of writing, they are exposed only through class components.
+This is an example of the implementation:
+
+```javascript
+class App extends React.Component {
+    codePushStatusDidChange(status) {
+        switch(status) {
+            case codePush.SyncStatus.CHECKING_FOR_UPDATE:
+                console.log("Checking for updates.");
+                break;
+            case codePush.SyncStatus.DOWNLOADING_PACKAGE:
+                console.log("Downloading update.");
+                break;
+            case codePush.SyncStatus.INSTALLING_UPDATE:
+                console.log("Installing update.");
+                break;
+            case codePush.SyncStatus.UP_TO_DATE:
+                console.log("Your app is up-to-date.");
+                break;
+            case codePush.SyncStatus.UPDATE_INSTALLED:
+                console.log("Update installed.");
+                break;
+        }
+    }
+
+    codePushDownloadDidProgress(progress) {
+        console.log(progress.receivedBytes + " of " + progress.totalBytes + " received.");
+    }
+}
+App = codePush(MyApp);
+export default App;
+```
+
+The `codePushStatusDidChange()` event is raised when the status of the update changes. It can assume one of the values exposed by the `SyncStatus` enumerator. The `codePushDownloadDidProgress()` event, instead, is raised when the download has started and, through the `progress` object, you can get the current status, in case you want to display a progress bar or another similar control.
+
+## Wrapping up
+
+Thanks to CodePush, you can be more agile in keeping your application up-to-date and deliver new features without having to redeploy a completely new package. Currently, App Center doesn't fully support the React Native for Windows implementation, but with a few workarounds you can still leverage it. I highly invite you to read [the whole README of the main repository](https://github.com/microsoft/react-native-code-push/blob/master/README.md), since it contains many additional information for advanced scenarios, like how to handle Store policies, how to manage multiple deployment assignments, etc.
```

Comment:
```suggestion
Thanks to CodePush, you can be more agile in keeping your application up-to-date and deliver new features without having to redeploy a completely new package. Currently, a few workarounds are needed for App Center to integrate with the React Native for Windows implementation. I highly recommend you to read [the whole README of the main repository](https://github.com/microsoft/react-native-code-push/blob/master/README.md), since it contains a lot of additional information for advanced scenarios, like how to handle Store policies, how to manage multiple deployment assignments, etc.
```